core/src/main/java/io/micronaut/core/annotation/AnnotationValue.java

```diff
@@ -1703,4 +1709,18 @@ private static <T extends Enum> Optional<T> convertToEnum(Class<T> enumType, Obj
             }
         }
     }
+
+    /**
+     * Creates {@link AnnotationClassValue} from the class name.
+     *
+     * @param className The class name
+     * @return AnnotationClassValue for given class name
+     */
+    private static AnnotationClassValue<?> getAnnotationClassValue(String className) {
+        Optional<Class<?>> theClass = ClassUtils.forName(className, null);
```

Comment:
The difference from previous version is that it was using classloader from `beanContext.getClassLoader()` and this will fallback to current thread classloader.
```
if (classLoader == null) {
                classLoader = Thread.currentThread().getContextClassLoader();
            }
            if (classLoader == null) {
                classLoader = ClassLoader.getSystemClassLoader();
            }
```
Not sure if this could cause other issues.